packages/turf-distance/index.ts

```diff
@@ -1,25 +1,33 @@
 import { Point } from "geojson";
 import { getCoord } from "@turf/invariant";
-import { radiansToLength, degreesToRadians, Coord, Units } from "@turf/helpers";
-
+import type { Coord, Datum, Units } from "@turf/helpers";
+import {
+  datums,
+  radiansToLength,
+  degreesToRadians,
+  convertLength,
+} from "@turf/helpers";
+import LatLon from "geodesy/latlon-ellipsoidal-vincenty.js";
 //http://en.wikipedia.org/wiki/Haversine_formula
 //http://www.movable-type.co.uk/scripts/latlong.html
 
 /**
- * Calculates the distance between two {@link Point|points} in degrees, radians, miles, or kilometers.
- * This uses the [Haversine formula](http://en.wikipedia.org/wiki/Haversine_formula) to account for global curvature.
+ * Calculates the distance between two {@link Point|points}.
+ * If a specific datum is passed, uses a geodesic ellipsoid calculation.
```

Comment:
I think you are technically correct here in that if the user passes any valid datum object, that the code as it is now will run `geodesicEllipsoidDistance`.  But that doesn't seem clean/clear.  Technically, I could possibly make up a datum and it will still run that one function right? But why? It seems like it would be better to accept the one expected ellipsoid datum, and throw an error otherwise.  Also see my comment below about exposing just a string parameter to specify the datum.